src/main/java/com/sequenceiq/cloudbreak/domain/Recipe.java

```diff
@@ -0,0 +1,129 @@
+package com.sequenceiq.cloudbreak.domain;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.persistence.Column;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+@Entity
+@Table(name = "recipe", uniqueConstraints = {
+        @UniqueConstraint(columnNames = { "account", "name" })
+})
+@NamedQueries({
+        @NamedQuery(
+                name = "Recipe.findForUser",
+                query = "SELECT r FROM Recipe r "
+                        + "WHERE r.owner= :owner"),
+        @NamedQuery(
+                name = "Recipe.findPublicInAccountForUser",
+                query = "SELECT r FROM Recipe r "
+                        + "WHERE (r.account= :account AND r.publicInAccount= true) "
+                        + "OR r.owner= :owner"),
+        @NamedQuery(
+                name = "Recipe.findAllInAccount",
+                query = "SELECT r FROM Recipe r "
+                        + "WHERE r.account= :account "),
+        @NamedQuery(
+                name = "Recipe.findByNameForUser",
+                query = "SELECT r FROM Recipe r "
+                        + "WHERE r.name= :name and r.owner= :owner "),
+        @NamedQuery(
+                name = "Recipe.findByNameInAccount",
+                query = "SELECT r FROM Recipe r WHERE r.name= :name and r.account= :account")
+})
+public class Recipe implements ProvisionEntity {
+
+    @Id
+    @GeneratedValue
```

Comment:
maybe we should specify a sequence generator here as well like for the other entities